### PR TITLE
Add LayerNormLSTMCell to RNN cell

### PR DESCRIPTION
**Pull request by qlzh727**
1. The change was based on contrib.LayerNormBasicLSTMCell, which is popular based on https://tf-contrib-analyzer.herokuapp.com.
2. The cell has been updated to align with keras and v2 standard, eg use keras LSTMCell as base class, get rid of variable scope, etc.
3. The tests have also been ported and updated with graph/eager mode.

### CODE REVIEW

File: tensorflow_addons/rnn/cell.py [hunk 1]
```diff
@@ -208,3 +208,183 @@ def get_config(self):
...
+
+    This class adds layer normalization and recurrent dropout to a LSTM unit.
+    Layer normalization implementation is based on:
+
+      https://arxiv.org/abs/1607.06450.
+
+    "Layer Normalization" Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton
+
+    and is applied before the internal nonlinearities.
+    Recurrent dropout is base on:
```
Comment (WindQAQ) [inline tensorflow_addons/rnn/cell.py]: base -> based

**Comment by qlzh727** [comment]
@WindQAQ, thanks for the question. My understanding is that the weights between each gates are separate and we only fuse them together so that we can do large matmul to improve performance. Since each gate carries different mathematical meaning (input, carry, forget, output), if we do normalization for the fused gates, it doesn't make sense for math, whereas the normalization for individual gates makes more sense to me.

That said, since it does seems to be different from the original paper, let me check with the engineer who implemented it originally.

**Comment by qlzh727** [comment]
As a side note, from the implementation in https://github.com/ryankiros/layer-norm/blob/master/torch_modules/LSTM_LN.lua#L30, which is from the original paper author, only the output is normalized, instead of individual gates. This is also different from the original formula in the paper, which makes me quite confused.

**Comment by qlzh727** [comment]
@ebrevdo and @drpng

**Comment by WindQAQ** [comment]
@qlzh727, Thanks for the investigation!

> As a side note, from the implementation in https://github.com/ryankiros/layer-norm/blob/master/torch_modules/LSTM_LN.lua#L30, which is from the original paper author, only the output is normalized, instead of individual gates. This is also different from the original formula in the paper, which makes me quite confused.

I found that the theano implementation in the same repo indeed applies layer norm on fused gate. To be more clear, it seems to match the formula stated in the paper.
https://github.com/ryankiros/layer-norm/blob/master/layers.py#L457-L475

**Comment by qlzh727** [comment]
Done. I have update the implementation to align with the original paper, even it doesn't make much sense to me. I don't have a e2e model to verify this, but since this matches to the reference in https://github.com/ryankiros/layer-norm/blob/master/layers.py#L457-L475, I will follow that implementation.

Also fixed the issue that bias is ignored when norm was on. I think that was a bug introduced by the original implementation, and have confirmed that with original author.

**Comment by qlzh727** [comment]
Sorry, seems that I was incorrectly rebase and pull my master branch, which somehow include my other PR into here. Should I revert this one and start with a new PR?

**Comment by seanpmorgan** [comment]
No worries, but yeah new PR is probably easier

**Comment by qlzh727** [comment]
Done. See https://github.com/tensorflow/addons/pull/210. I am closing this PR now.
